lib/manageiq/api/client/collection.rb

```diff
@@ -42,6 +43,58 @@ def self.subclass(name)
             const_set(klass_name, klass)
           end
         end
+
+        def ar_search(options)
+          search(query_parameters_from_ar_query_options(options))
+        end
+
+        private
+
+        def query_parameters_from_ar_query_options(options)
+          api_params = {}
+          [:offset, :limit].each { |opt| api_params[opt] = options[opt] if options[opt] }
+          api_params[:attributes] = options[:select].join(",") if options[:select].present?
+          if options[:where]
+            api_params[:filter] ||= []
+            conditional_parameter_from_ar_query_options(api_params[:filter], "=", options[:where])
+          end
+          if options[:not]
+            api_params[:filter] ||= []
+            conditional_parameter_from_ar_query_options(api_params[:filter], "!=", options[:not])
+          end
+          if options[:order]
+            order_parameters_from_ar_query_options(api_params, options[:order])
+          end
+          api_params
+        end
+
+        def conditional_parameter_from_ar_query_options(filter_param, condition, ar_option)
```

Comment:
This method name reads as a noun (i.e. it's going to return the conditional parameter from AR query options), but it actually modifies the filter_param.  That makes the name very confusing (I actually thought the caller was throwing away the return value and was about to comment there when I saw this).

Prefer noun-named-methods to have no side effects and to return whatever the noun is.  Then create a verb-named-method that uses the noun-named-method to modify the variable or change state or whatever (or just inline it if a whole separate verb-named-method feels like overkill)